gui/src/CMakeLists.txt

```diff
@@ -14,7 +14,28 @@ target_sources(
           sidebarmanager.hh
           view.cc
           view.hh
-          workinterface.hh)
+          workinterface.hh
+          ruler.cc
+          ruler.hh
+          rulerstrategies.cc
+          rulerstrategies.hh)
+
+if(ENABLE_BOOST_TEST)
+  # Add a ruler code to library for unit testing
+  add_library(ruler_lib)
+  target_sources(
+          ruler_lib
+          PRIVATE ruler.cc
+                  ruler.hh
+                  rulerstrategies.cc
+                  rulerstrategies.hh)
+  target_link_libraries(
+          ruler_lib
+          PRIVATE project_options project_warnings
+          PUBLIC util
+                 PkgConfig::gtk
+                 Boost::system)
```

Comment:
Oops... found the `add_test` call in `gui/CMakeLists.txt`. However, the test does not get run or built, so something is still missing.

Also, since you are putting the ruler in a library, why are you not (also) using that library in the gui? And move it to its own directory?

Reply:
Not sure what changed, but ruler tests are now executing...